runtime/domain.c

```diff
@@ -1081,7 +1162,13 @@ static void domain_terminate()
     Caml_state->critical_section_nesting = 0;
     acknowledge_all_pending_interrupts();
   }
-  caml_plat_unlock(&domain_self->roots_lock);
+
```

Comment:
Worth encasing the following block in a check for `backup_thread_running` ?

Reply:
Since this is in `domain_terminate`, which is only called in the case a domain is spawned explicitly, the backup thread is definitely running. 

I'll add an assertion `Assert (domain_sef->backup_thread_running)`.

Reply:
This assertion triggered a bug in domain and backup thread termination. The latest commit makes the main domain thread wait for the backup thread to terminate before terminating itself.